packages/parser-typescript-config/src/parser.ts

```diff
@@ -67,6 +71,28 @@ export default class TypeScriptConfigParser extends Parser {
         try {
             config = JSON.parse(fetchEnd.response.body.content);
 
+            const originalConfig = cloneDeep(config);
+
+            try {
+                config = finalConfig(config, resource);
+            } catch (err) {
+                if (err.code === ErrorCodes.circular) {
```

Comment:
This checks is also in Babel. Can't it be refactored and maybe added to the `Parser` class so it's available in all `Parsers`?